internal/network/dual_stack_validations.go

```diff
@@ -0,0 +1,61 @@
+package network
+
+import (
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+)
+
+// Verify if the constrains for dual-stack machine networks are met:
+//   * there are exactly two machine networks
+//   * the first one is IPv4 subnet
+//   * the second one is IPv6 subnet
+func VerifyMachineNetworksDualStack(networks []*models.MachineNetwork, isDualStack bool) error {
+	if !isDualStack {
+		return nil
+	}
+	if len(networks) != 2 {
+		return errors.Errorf("Expected 2 machine networks, found %d", len(networks))
+	}
+	if !IsIPV4CIDR(string(networks[0].Cidr)) {
+		return errors.Errorf("First machine network has to be IPv4 subnet")
+	}
+	if !IsIPv6CIDR(string(networks[1].Cidr)) {
+		return errors.Errorf("Second machine network has to be IPv6 subnet")
```

Comment:
Have we given any thought to creating custom `error` types?

Reply:
Never thought about this honestly. How would that look like ?

Reply:
I may be missing something from @djzager's idea but, considering that we don't check the type of any of these errors, I'm unsure how much value custom error types would bring. Perhaps reduce duplication of error messages? Could we use constants for some those?

Reply:
I'd keep those ones as they are. The strings are only for humans to read them, we don't automate anything based on errors and based on my previous experience, it may happen that we start changing those messages very soon based on the users' understanding of them

Reply:
More of a curiosity question than trying to ask for a change.

The only thing that stands out to me as a con to using `errors.Errorf` was that all of our tests are just looking for some hardcoded string. The pro I've experienced is that it makes it easy to find where a failure is coming from when the strings are hardcoded this way :sunglasses: .

This can be resolved.